hypervisor/debug/profiling.c

```diff
@@ -112,18 +112,25 @@ static void profiling_enable_pmu(void)
 		__func__, MSR_IA32_EXT_APIC_LVT_PMI, lvt_perf_ctr);
 
 	if (ss->guest_debugctl_value != 0U) {
-		/* Set the VM Exit MSR Load in VMCS */
+		/* Merge the msr vmexit loading list with HV */
 		if (ss->vmexit_msr_cnt == 0) {
-			ss->vmexit_msr_cnt = 1;
-			ss->vmexit_msr_list[0].msr_idx
+			struct acrn_vcpu *vcpu = get_ever_run_vcpu(get_cpu_id());
+
+			ss->vmexit_msr_cnt = 1 + MSR_AREA_COUNT;
```

Comment:
DId not quite understand. You defined MAX_MSR_LIST_NUM etc., then why use 1 here?

Reply:
+1 indicates we add IA32_DEBUGCTL MSR on top of the VMExit MSRs. Below the vmexit_msr_list is defined to combine all of them.
struct msr_store_entry vmexit_msr_list[MAX_PROFILING_MSR_LIST_NUM + MAX_HV_MSR_LIST_NUM];
Currently, only one MSR is added for profiling but we may add more later. That's why we allocate enough space in MAX_PROFILING_MSR_LIST_NUM.

Reply:
In this case, why we not define MAX_MSR_LIST_NUM to 1 (this way, we can eliminate "1")? we can change later on.

BTW, you store HV MSRs after PROFILING MSR. If we change the order, we can avoid "1" too.